src/main/java/com/zufar/onlinestore/user/api/UserService.java

```diff
@@ -19,11 +22,16 @@ public class UserService implements UserApi {
 
     private final UserRepository userCrudRepository;
     private final UserDtoConverter userDtoConverter;
+    private final AuthorityService authorityService;
+    private final UserDataValidator userDataValidator;
 
     @Override
     public UserDto saveUser(final UserDto userDto) {
         UserEntity userEntity = userDtoConverter.toEntity(userDto);
+        userDataValidator.validate(userEntity);
```

Comment:
You should create A Custom Validator Annotation with Spring like that:
https://reflectoring.io/bean-validation-with-spring-boot/

Reply:
Done